Tests/NIOSSHTests/UserAuthenticationStateMachineTests.swift

```diff
@@ -720,4 +730,27 @@ final class UserAuthenticationStateMachineTests: XCTestCase {
         // Let's say we got a success. Happy path!
         XCTAssertNoThrow(try stateMachine.receiveUserAuthSuccess())
     }
+
+    func testCertificateClientAuthFlow() throws {
+        let delegate = InfinitePrivateKeyDelegate()
```

Comment:
This test is wrong: you aren't using your new delegate.

Reply:
yep, fixed!